main.tf

```
@@ -69,11 +69,14 @@ resource "aws_security_group_rule" "egress" {
 # The name "primary" is poorly chosen. We actually mean standalone or regional.
 # The primary cluster of a global database is actually created with the "secondary" cluster resource below.
 resource "aws_rds_cluster" "primary" {
-  count                               = local.enabled && local.is_regional_cluster ? 1 : 0
-  cluster_identifier                  = var.cluster_identifier == "" ? module.this.id : var.cluster_identifier
-  database_name                       = var.db_name
+  count              = local.enabled && local.is_regional_cluster ? 1 : 0
+  cluster_identifier = var.cluster_identifier == "" ? module.this.id : var.cluster_identifier
+  database_name      = var.db_name
+  # manage_master_user_password must be `null` or `true`. If it is `false`, and `master_password` is not `null`, a conflict occurs.
+  manage_master_user_password         = var.manage_admin_user_password == false ? null : var.manage_admin_user_password
```

```suggestion
  manage_master_user_password         = var.manage_admin_user_password ? null : var.manage_admin_user_password
```

---

How come it can't be this ?

```suggestion
  manage_master_user_password         = var.manage_admin_user_password
```

---

> How come it can't be this ?

See the comment above it:
```
# manage_master_user_password must be `null` or `true`. If it is `false`, and `master_password` is not `null`, a conflict occurs.
```

When it was just that previously, it broke the CI tests, so it had to be more explicit. The OG PR from @salemgolemugoo had the `default` as `null`, but that was requested to be changed to `false` [here](https://github.com/cloudposse/terraform-aws-rds-cluster/pull/221#discussion_r1682877763), necessitating the need to be explicit.

---

@gberenice it can't be that, because it's checking for falsiness, not truthiness.
It could be this though:
```
  manage_master_user_password         = var.manage_admin_user_password ? var.manage_admin_user_password : null

```

---

@pazaan sorry, my bad. Of course, it should be like your latest comment. Mind updating?

---

@gberenice 
> Mind updating?

I just realised that if we went this way, it would need to be:

```
  manage_master_user_password         = try(var.manage_admin_user_password ? var.manage_admin_user_password : null, null)
```

Because if the end user sets `var.manage_admin_user_password` to `null`, `manage_master_user_password         = var.manage_admin_user_password ? var.manage_admin_user_password` would throw the following error:
```
│ Error: Null condition
│ 
│   on <console-input> line 1:
│   (source code not available)
│ 
│ The condition value is null. Conditions must either be true or false.
```

So, either:
```
  manage_master_user_password         = var.manage_admin_user_password == false ? null : var.manage_admin_user_password
```
or
```
  manage_master_user_password         = try(var.manage_admin_user_password ? var.manage_admin_user_password : null, null)
```

---

@pazaan `null` is an invalid value here as we expect a boolean. If you're concerned about this case, you can set `nullable` to false, so Terraform will not accept null as a valid value and will raise an error:
```hcl
variable "manage_admin_user_password" {
  type        = bool
  default     = false
  description = "Set to true to allow RDS to manage the master user password in Secrets Manager. Cannot be set if master_password is provided"
  nullable    = false
}
```
Does that make sense?

---

@gberenice Yes, setting `nullable` to `false` is much more elegant 👍🏻